framework/src/main/java/org/checkerframework/framework/util/ComponentFinderUtil.java

```diff
@@ -0,0 +1,83 @@
+package org.checkerframework.framework.util;
+
+import org.checkerframework.common.basetype.BaseTypeChecker;
+
+/**
+ * Util class to find components (e.g. visitors and factories) following the naming convention
+ * reflectively. For example, ABCChecker's default naming for visitor is "ABCVisitor". If a visitor
+ * class with that name exists, then instantiate it and returns. Otherwise try to find the super,
+ * and finally uses the {@code defaultGetter} callback to get a default value if all attempt fails.
+ */
+public class ComponentFinderUtil {
+
+    /**
+     * A single method interface for getting default type for components by method reference.
+     *
+     * @param <DefaultType> the type for the default value
+     */
+    public interface DefaultGetter<DefaultType> {
+
+        /**
+         * The logic for getting the default value if all attempt fails
+         *
+         * @param checker a checker as the argument of constructor
+         * @return the default value
+         */
+        DefaultType getDefault(BaseTypeChecker checker);
+    }
+
+    /**
+     * Find a component named with the checker naming convention.
+     *
+     * @param <T> type of the component
+     * @param checker the current checker
+     * @param replacement the component suffix, e.g. "Visitor" or "AnnotatedTypeFactory"
```

Comment:
If this is a suffix, why not name it "suffix".  The documentation doesn't indicate why it is named "replacement".  The fact that replacement is done within a method called by this one is an implementation detail.